stock_picking_invoicing/wizards/stock_invoice_onshipping.py

```diff
@@ -0,0 +1,553 @@
+        :param values: dict
+        :return: dict
+        """
+        line = self.env["account.move.line"].new(values.copy())
+        line._onchange_product_id()
+        new_values = line._convert_to_write(line._cache)
+        if price_unit:
+            new_values["price_unit"] = price_unit
+        # Ensure basic values are not updated
+        values.update(new_values)
+        return values
+
+    def _get_invoice_line_values(self, moves, invoice_values, invoice):
+        """
+        Create invoice line values from given moves
+        :param moves: stock.move
+        :param invoice: account.move
+        :return: dict
+        """
+        name = ", ".join(moves.mapped("name"))
+        move = fields.first(moves)
+        product = move.product_id
+        fiscal_position = self.env["account.fiscal.position"].browse(
+            invoice_values["fiscal_position_id"]
+        )
+        partner_id = self.env["res.partner"].browse(invoice_values["partner_id"])
+        categ = product.categ_id
+        inv_type = invoice_values["move_type"]
+        if inv_type in ("out_invoice", "out_refund"):
+            account = product.property_account_income_id
+            if not account:
+                account = categ.property_account_income_categ_id
+        else:
+            account = product.property_account_expense_id
+            if not account:
+                account = categ.property_account_expense_categ_id
+        account = move._get_account(fiscal_position, account)
+        quantity = 0
+        move_line_ids = []
+        for move in moves:
+            qty = move.product_uom_qty
+            loc = move.location_id
+            loc_dst = move.location_dest_id
+            # Better to understand with IF/ELIF than many OR
+            if inv_type == "out_invoice" and loc.usage == "customer":
+                qty *= -1
+            elif inv_type == "out_refund" and loc_dst.usage == "customer":
+                qty *= -1
+            elif inv_type == "in_invoice" and loc_dst.usage == "supplier":
+                qty *= -1
+            elif inv_type == "in_refund" and loc.usage == "supplier":
+                qty *= -1
+            quantity += qty
+            move_line_ids.append((4, move.id, False))
+        taxes = moves._get_taxes(fiscal_position, inv_type)
+        price = moves._get_price_unit_invoice(inv_type, partner_id, quantity)
+        line_obj = self.env["account.move.line"]
+        values = line_obj.default_get(line_obj.fields_get().keys())
+        values.update(
+            {
+                "name": name,
+                "account_id": account.id,
+                "product_id": product.id,
+                "product_uom_id": product.uom_id.id,
+                "quantity": quantity,
+                "price_unit": price,
+                "tax_ids": [(6, 0, taxes.ids)],
+                "move_line_ids": move_line_ids,
+                "move_id": invoice.id,
+            }
+        )
```

Comment:
Same comment as before : account_id, product_uom_id and tax_ids are already computed by odoo onchange.